### PR TITLE
Add git hook file for running spotless as pre-commit

### PR DESCRIPTION
**Pull request by magicus**
This PR contains a pre-commit hook for git, that will run spotlessApply before committing. This will make you avoid committing code that will not build with gradle.

The hook is optional to use. If you want to use it, you must tell git to pick up hooks from the directory containing the hook. Run this from your repo root: `git config core.hooksPath utils/git-hooks`

### CODE REVIEW

**Comment by kristofbolyai** [comment]
This does not seem to work as intented? When using the hook, it only formats the code after the commit, meaning that it is not included in the commit. 
image
image

**Comment by kristofbolyai** [comment]
Same behavior with CLI, my test command was: `git commit -a -m "Test"`

**Comment by magicus** [comment]
*sigh* It seemed to work for me but in all honesty I don't really think I actually exercised it properly. :(

Ok, back to the drawing table...

**Comment by magicus** [comment]
I have now verified that it works both with staged files, and if you have unstaged files and commit with `-a`.

**Comment by kristofbolyai** [comment]
It does not seem to work correctly when using Intellij IDEA's Github modal. Also it does not seem to work with git CLI. It does apply the spotless formatting, but does not include it in the commit. Currently it formats, commits unformatted, and you can commit again, this time the formatted version.

**Comment by kristofbolyai** [comment]
image
image
image

**Comment by magicus** [comment]
That is really odd. I tested it with the git CLI, and it worked for me.

**Comment by magicus** [comment]
I'm looking at your screen shots and are trying to figure out what you're showing. Have you tried committing a piece of code that is not accepted by spotless, and then after committing it, looking at what was actually committed? That was what I did (admittedly not from IntelliJ -- maybe it does not respect hooks?), and it worked fine.

**Comment by magicus** [comment]
There is a "Run git hooks" checkbox in the IntelliJ commit screen. Make sure it is checked. I just tested this, and it worked fine.

Have your run the git command for setting up the hook directory?

**Comment by kristofbolyai** [comment]
I did have the hooks added and turned on. Let me try again.

**Comment by kristofbolyai** [comment]
I did not have the updates pulled for some reason. 🤡

**Review by kristofbolyai — APPROVED** [review]
